### DISCUSSION TITLE
HSM

### DISCUSSION THREAD

**Discussion by Dluxfrance** [polhenarejos/pico-hsm#21] [Q&A] [answered]
hi
just found this great project
I downloaded and build successfully
but cant use scsh - it gives me error  "Device authentication failed"
any idea?
BTW if build using Docker - it build version 2.6 and its working fine with scsh 
thanks
David

**Comment by polhenarejos** [comment, marked as answer]
Did you use `pico-hsm-tool.py` to initialize it? If not, try to:
```
python3 tools/pico-hsm-tool.py initialize
```
and follow the instructions.

**Reply by Dluxfrance** [reply]
pico-hsm-tool working great, 
only  scsh not 
i followed this doc: https://github.com/polhenarejos/pico-hsm/blob/master/doc/scs3.md 
got the above error

**Reply by polhenarejos** [reply]
Have you patched correctly SCS3 to recognize Pico HSM?
Paste here what is displayed in the `Trace` tab.

**Reply by Dluxfrance** [reply]
yes i patche it
it also working ok with version 2.6 
but on 3.4 while trying to load the key manager i get:

GPError: KeyManager (CRYPTO_FAILED/0) - "Device authentication failed" in /home/david/CardContact/scsh3/keymanager/keymanager.js#205
    at /home/david/CardContact/scsh3/keymanager/keymanager.js#205
    at /home/david/CardContact/scsh3/keymanager/keymanager.js#42
    at /home/david/CardContact/scsh3/keymanager/keymanager.js#2461

**Reply by polhenarejos** [reply]
From v2.6 to v3.4 the PKI changed root CV certificates. SCS3 must be patched with new certificates. Can you check your scs3 tool is patched with correct certificates? 

New certificate is
```
ESPICOHSMCA00001: new CVC(new ByteString("7F218201BA7F4E8201725F290100421045535049434F48534D434130303030317F4982011D060A04007F000702020202038120FFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFF8220FFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFC83205AC635D8AA3A93E7B3EBBD55769886BC651D06B0CC53B0F63BCE3C3E27D2604B8441046B17D1F2E12C4247F8BCE6E563A440F277037D812DEB33A0F4A13945D898C2964FE342E2FE1A7F9B8EE7EB4A7C0F9E162BCE33576B315ECECBB6406837BF51F58520FFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC6325518641046A82C0A4FEAF41D6A1336AE7E992D81AD4F827929145DD0D777E1AB63D7E3325C8F7DAC0F74B6EAE13A72F6366777EC133AC5C28F456868E5F2C315044EB54EF8701015F201045535049434F48534D434130303030317F4C12060904007F0007030102025305C0000000005F25060202000801085F24060203000801085F3740601E974F57DDE060875FE6121AEF5BC02E10FC655311C7A32CA822FD18E53A80298EDC56E0D5EBF38FB470DC12987B1600AE91A0ADB5B22C4D80080782E278AD", HEX))
```

**Reply by Dluxfrance** [reply]
working!!!!
thanks
